articles/openshift/oshift-how-disable-weak-cipher-suites-enable-hsts.md

```diff
@@ -0,0 +1,114 @@
+---
+title: How to disable weak cipher suites and enable HSTS in OpenShift
+description: Provides instructions regarding disabling insecure cipher suites and enabling HSTS for OpenShift clusters
+services: openshift
+author: Ben Bacon
+reviewer: 
+lastreviewed: 24/06/2020
+toc_rootlink: How To
+toc_sub1: 
+toc_sub2:
+toc_sub3:
+toc_sub4:
+toc_title: Disable weak cipher suites and enable HSTS
+toc_fullpath: How To/oshift-how-disable-weak-cipher-suites-enable-hsts.md
+toc_mdlink: oshift-how-disable-weak-cipher-suites-enable-hsts.md
+---
+
+# How to disable weak cipher suites and enable HSTS
+
+## Overview
+
+This article outlines steps to increase the security of externally exposed routes from your OpenShift cluster, through the use of more secure cipher suites and HSTS.
+
+### Intended audience
+
+OpenShift developers who have created and deployed services into OpenShift and created and exposed secure routes for those services.
+
+## Disable weak cipher suites
+
+Cipher suites perform an important role in determining the security of a TLS connection. Over time, vulnerabilities have been identified in the algorithms used by cipher suites, which has resulted in published methods of decrypting and manipulating encrypted traffic. It's therefore generally recommended to use as modern a cipher suite as possible.
+
+### UKCloud for OpenShift v3.x
+
+#### Data plane
+
+To provide extended client compatibility, the OpenShift router supports cipher suites from TLS 1.0 - TLS 1.2 by default. Some of these cipher suites (especially older ones) are vulnerable to man-in-the-middle (MITM) attacks.
+
+If exposed secure routes are only being accessed from modern clients (that support TLS 1.2 and above),  you can disable the weak cipher suites by adding the following environment variable to any router `DeploymentConfig`:
+
+```
+oc set env dc/router ROUTER_CIPHERS=modern -n default
+```
+
+This triggers a rollout of the `DeploymentConfig`. Going forward, the new router pods will only accept cipher suites from TLS 1.2.
+
+#### Control plane
+
+By default, the OpenShift control plane (web console) only supports TLS 1.2 ciphers, although these do not provide forward secrecy.
+
+We can restrict the ciphers to only those that provide forward secrecy. To do this, raise a Service Request with the UKCloud OpenShift Support team via the [My Calls](https://portal.skyscapecloud.com/support/ivanti) section of the UKCloud Portal.
+
+### UKCloud for OpenShift v4.x
+
+#### Data plane
+
+There are no weak ciphers supported by the OpenShift `IngressController` since by default it supports TLS 1.2 cipher suites with forward secrecy. It is possible to further restrict the ciphers by modifying the default `IngressController` object, specifically the value of `IngressController.spec.tlsSecurityProfile`:
```

Comment:
Could we say what the value should be? The command below will just allow the customer to edit the config, I'm not clear what the value should be for the IngressController.spec.tlsSecurityProfile

Reply:
I've added an example is this clearer?